lib/clickhousex/codec/json.ex

```diff
@@ -71,11 +73,30 @@ defmodule Clickhousex.Codec.JSON do
     Enum.map(value, &to_native(type, &1))
   end
 
+  defp to_native(<<"Tuple(", types::binary>>, value) do
+    types
+    |> String.replace_suffix(")", "")
+    |> then(&Regex.scan(@types_regex, &1, capture: :all_names))
+    |> Enum.with_index()
+    |> Enum.reduce(%{}, fn {[column, type], index}, acc ->
+      Map.put(acc, column, to_native(type, Enum.at(value, index)))
+    end)
+  end
+
+  defp to_native(<<"Map(", map_types::binary>>, value) do
+    map_types = String.replace_suffix(map_types, ")", "")
+    [key_type, value_type] = String.split(map_types, ", ", parts: 2)
+
+    value
+    |> Enum.map(fn {key, value} -> {to_native(key_type, key), to_native(value_type, value)} end)
+    |> Map.new()
```

Comment:
Maybe `reduce` will be better?
```
Enum.reduce(value, %{}, fn {key, value}, acc -> 
  Map.put(acc, to_native(key_type, key), to_native(value_type, value)) 
end)

```

Reply:
Fixed. I had to remove conversion to map for tuples because its not guaranteed that tuple will have named columns